src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs

```diff
@@ -64,21 +75,68 @@ public Context CreateContext(IFeatureCollection contextFeatures)
                 // Non-inline
                 LogRequestStarting(httpContext);
             }
-
             if (diagnoticsEnabled)
             {
+#if !NET451
+                context.Activity = StartActivity(httpContext);
+#endif
                 // Non-inline
                 RecordBeginRequestDiagnostics(httpContext, startTimestamp);
             }
 
-            // Create and return the request Context
-            return new Context
+            // Fill and return the request Context
+            context.HttpContext = httpContext;
+            context.Scope = scope;
+            context.StartTimestamp = startTimestamp;
+            return context;
+        }
+
+#if !NET451
+        private Activity StartActivity(HttpContext httpContext)
```

Comment:
I would like this code to either follow the lead of the [outgoing HTTP code](https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L33-L147) or we have good reasons for the differences (we can change the outgoing code if we feel we made a mistake).

On of the big things that the outgoing HTTP code did was NOT Try to reuse the existing events.  It kept them, but made it clear in the source code at least that they are deprecated.   This allows the events to use better names (making activities have 'Start' and 'Stop' suffixes) that will be consistent with all future activities.     The code is reasonably nicely segregated to make it easy to remove the old events after a time.   

It also starts conventions about filtering.    It assumes there are three levels of interesting filtering

1.  Off completely
2. Propagate (if something else turned on activity tracking, continue to do so) don't log events however 
3. On (or sampling) - low overhead   Turn on activity tracking even if it was off (that is initiate a new sample) only log Stop events (you want to log as little as possible, and Stop is the obvious choice since you know the most at that point)
4. On  with start events - everything in (3) plus start events.   Gives the receiver the most control/information.   

It achieves this by 

- if DiagnosticSource.IsEnabled() (no parameters) return true then we are in propagate.
- if DiagnosticSource.IsEnabled(ActivityName, request) returns true then we are ON (you get stops, you create an activity even if there was no parent) 
- if DiagnosticSource.IsEnabled(ActivityStartName) is true then we also log Start as well as Stop events. 

I would like us to 
1. Come to a consensus on what the right semantics are (I am suggesting the semantics above, both for compat as well as for how to turn on various levels.).
2. Insure that both of these implementations conform to what we decided.  

Comments?